### PR TITLE
Bug #1581 subtraction errors

### LINKED ISSUE #1581: Errors in subtraction branch
### Description

Required: insert bug description here

* Major error was a simple copy-paste one, where `xASL_im_ASLSubtractionAverage` erroneously computes PWI3D from Control4D. Fixed in b50714f023b6b28aaad2a412bea09818e58dd8ec
* Also, we forgot to create mean_control which is then probably skipped in `xASL_wrp_RegistrationASL`. Fixed in 3ecfa3bd87d0cd4edb250be068a4198f1786f87d & bugfixed in 57826cae612618041d6157e424430878820a4e60
* Further discovered but unrelated errors will be fixed in #1588 

### Tasks

- [x] GE_3Dspiral_T1prefilled_Cyst_M0_noFLAIR_2ndRun_LoQ	
- [x] Philips_2DEPI_FLAIR_DisabledQuantification_M0_LoQ	- OK
- [x] Philips_2DEPI_FLAIR_Segm_PWI4D_Quant_Parameters_LoQ	- minor error
- [x] Philips_3DGRASE_FLAIR_M0_2ndRun_HiQ	- large error
- [x] Philips_3DGRASE_MultiSessions_T1wAlreadyDone_LoQ	- large error 

### We need to fix despiked_ASL4D for the following to run - #1543
- [x] Philips_2DEPI_Segmentation_2D_T2_SPM_HiQ	- large error 100%
- [x] Philips_2DEPI_FEAST_Longitudinal_FLAIR_LGA_ASL_PartlyDone_HiQ
- [x] Philips_2DEPI_Lesion_NoFLAIR_NoM0_HiQ
- [x] Siemens3DGRASE_Structural_AlreadyDone_HiQ
- [x] Siemens_2DEPI_noBsup_FLAIR_LoQ - #1543 despiked JSON

### How to test
- [x] Re-run the entire test again

### Release notes

Nothing. Report with #997

### PR DESCRIPTION
**Pull request by jan-petr**
### Linked issue

Close #1581

### CODE REVIEW

File: Modules/Module_Processing/Module_Population/xASL_wrp_CreatePopulationTemplates.m [hunk 1]
```diff
@@ -257,33 +257,33 @@ function xASL_wrp_CreatePopulationTemplates(x, bSaveUnmasked, bCompute4Sets, Spe
...
 	PreFixList{end+1}   = 'ATT';                                TemplateNameList{end+1}  = 'ATT';              SessionsExist(end+1)    = 1;
     PreFixList{end+1}   = 'FoV';                                TemplateNameList{end+1}  = 'FoV';              SessionsExist(end+1)    = 1;
     
-    PreFixList{end+1}   = '4V_MAP';                             TemplateNameList{end+1}  = '4V_MAP';           SessionsExist(end+1)    = 0;
-    PreFixList{end+1}   = '4V';                                 TemplateNameList{end+1}  = '4V';               SessionsExist(end+1)    = 0;
-    PreFixList{end+1}   = 'CoW_MAP';                            TemplateNameList{end+1}  = 'CoW_MAP';          SessionsExist(end+1)    = 0;
-    PreFixList{end+1}   = 'HEMI';                               TemplateNameList{end+1}  = 'HEMI';             SessionsExist(end+1)    = 0;    
-    PreFixList{end+1}   = 'TASL';                               TemplateNameList{end+1}  = 'TASL';             SessionsExist(end+1)    = 0;        
-	PreFixList{end+1}   = 'Tex';                                TemplateNameList{end+1}  = 'Tex';              SessionsExist(end+1)    = 1;
```
Comment (jan-petr) [inline Modules/Module_Processing/Module_Population/xASL_wrp_CreatePopulationTemplates.m:285]: Are we sure we don't need Tex and ABV templates? They might be useful!

**Review by jan-petr — COMMENTED** [review]
One question

File: Modules/Module_Processing/Module_Population/xASL_wrp_CreatePopulationTemplates.m (hunk 1 above)
Comment (HenkMutsaerts) [reply Modules/Module_Processing/Module_Population/xASL_wrp_CreatePopulationTemplates.m:285]: Reverted in 5053f41f584b00712001dac4ab90da523a440f13. They give messages to the user that it didn't find these files, so we should probably suppress these messages, let's think about this later.

**Review by HenkMutsaerts — APPROVED** [review]
See my answer to your last question and my last commit

File: Modules/Module_Processing/Module_Population/xASL_wrp_CreatePopulationTemplates.m (hunk 1 above)
Comment (jan-petr) [reply Modules/Module_Processing/Module_Population/xASL_wrp_CreatePopulationTemplates.m:285]: Agree. Issue 1592 created.

**Pull request merged**